e2e/deleteapi_test.go

```diff
@@ -18,7 +19,9 @@ func TestDeleteAPI(t *testing.T) {
 	deletedPods := make(map[string]struct{})
 	for _, pod := range apiPods.Items {
 		if err := client.CoreV1().Pods("kube-system").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{}); err != nil {
-			t.Fatalf("error deleting api-server pod: %v", err)
+			if _, ok := err.(http2.GoAwayError); !ok {
```

Comment:
I know this is painful, but I think the full fix here is:

1. Put the for loop inside a `wait.Poll()` (keep the `List()` outside the poll to make sure only delete the original pods instead of running in circles)
2. Ignore both `http2.GoAwayError` and `errors.IsNotExists()` (to make sure the poll is idempotent), `return false, err` on other errors.
3. After the for loop, `return len(deletedPods) == len(apiPods.Items), nil`

Reply:
i used retry... already imported into the test suite...

Reply:
trying out wait.Poll...